File: Python3/Pascal's Triangle.py
# 40ms, 100.0%
class Solution:
    def generate(self, numRows):
        res = [[1], [1, 1]]
        if numRows < 3: return res[: numRows]
        for _ in range(2, numRows):
            res.append([1, 1])
            for i in range(1, len(res[-2])):
                res[-1].insert(i, res[-2][i - 1] + res[-2][i])
        return res
    

File: Go/Pascal's Triangle.go
// 0ms, 100.00%; 2MB, 100.00%
func generate(numRows int) [][]int {
	tri := make([][]int, numRows, numRows)
	for i := 0; i < numRows; i++ {
		tri[i] = make([]int, i+1, i+1)
		tri[i][0], tri[i][i] = 1, 1
		for j := 1; j < i; j++ {
			tri[i][j] = tri[i-1][j-1] + tri[i-1][j]
		}
	}
	return tri
}
